vignettes/estimate_from_individual_data.Rmd

```diff
@@ -0,0 +1,321 @@
+```{r}
+# Get times of death for fatal outcomes
+death_times <- (case_onsets)[type_outcome == "D"] + outcome_death
+
+# Create a single data.frame with event types
+events <- data.frame(
+  dates = c(case_onsets, death_times),
+  event = c(rep("cases", length(case_onsets)),
+            rep("deaths", length(death_times)))
+)
+
+# Use incidence2 to calculate counts of cases and deaths by day
+counts <- incidence2::incidence(events,
+                                date_index = "dates",
+                                groups = "event",
+                                complete_dates = TRUE)
+
+# Pivot incidence object to get data.frame with counts for cases and deaths
+df <- counts %>%
+  tidyr::pivot_wider(names_from = event,
+                     values_from = count,
+                     values_fill = 0) %>%
+  dplyr::rename(date = date_index)
+
+cfr_static(df, delay_death)
+```
+Hence in this particular simulation, the `cfr_static()` method recovers the correct CFR of 10% (95% CI: 8.3-12.0%), whereas the filtering method produces a biased estimate of 26.4% (17.6-37.0%).
+
+## Deaths reported but not recoveries
+
+In an extreme scenario where recoveries are not reported, then we effectively have the values of `outcome_recovery` generated from a distribution with an infinite mean, and the above methods will still apply, with the same bias for the filtering approach. In particular, we would expect:
+
+$$
+E(\text{Total deaths})/ [ E(\text{Total deaths})+E(\text{Total recoveries}) ] \rightarrow 1 \\
+\text{as } E(\text{Total recoveries}) \rightarrow 0
+$$
+And hence the calculated CFR to incorrectly converge to 1 as the proportion of recoveries reported declines to 0.
+
+## Only total deaths reported
+
+In some situations, we may have a time series of cases but not deaths. However, we can still use the earlier calculation to derive an unbiased CFR:
+$$
+E(\text{Total deaths}) = p \sum_{t} \sum_j f_{j} I_{t-j} \\
+E(p) = \frac{\text{Total deaths}}{\sum_{t} \sum_j f_{j} I_{t-j} }
+$$
+We can do this in _cfr_ using the `estimate_outcomes()` function to calculate the expected number of cases with known fatal outcomes in the above denominator:
+```{r}
+# Calculate total deaths and total cases
+total_deaths <- sum(df$deaths)
+total_cases <- sum(df$cases)
+
+# Create data.frame with cases over time only
+df_case <- df
+df_case$deaths <- 0
+
+# Calculate the expected number of known fatal outcomes over time
+e_outcomes <- estimate_outcomes(df_case, delay_death)
+
+# Calculate the CFR
+total_deaths / (total_cases * tail(e_outcomes$u_t, 1))
+```
+As before, this produces a point estimate of 10.4%, because it is the same underlying method: the internal function code in `cfr_static()`, which calls `estimate_outcomes()`, also tallies up total death. An arguably more elegant approach is to therefore add total deaths to the case data.frame, and use as an input for `cfr_static()`:
```

Comment:
so this is the equivalent of saying that we know the total number of deaths by the end of the outbreak? it's cool! I think maybe we should add this info to the vignette too as it might help visualise the concept for readers

Reply:
Yep - internally the functions don't actually require the death timseries for the static calculation